micro-manager.py

```diff
@@ -38,7 +44,25 @@ def slice_tensor(a):
 
 # Define Gauss points on entire domain as coupling mesh
 couplingsample = domain.sample('gauss', degree=2)  # mesh located at Gauss points
-vertex_ids = interface.set_mesh_vertices(writeMeshID, couplingsample.eval(geom))
+
+coords_global = couplingsample.eval(geom)
+nv_global, _ = coords_global.shape
+nv_local = int(nv_global / size)
+
+coupling_coords = []
+# All processes except the last process get equal number of vertices
+if rank < size - 1:
+    for i in range(rank * nv_local, (rank + 1) * nv_local):
+        coupling_coords.append(coords_global[i])
+
+# Last process gets its share and remaining vertices if any
+if rank == size - 1:
+    for i in range(rank * nv_local, nv_global):
+        coupling_coords.append(coords_global[i])
+
+coupling_coords = np.array(coupling_coords)
```

Comment:
As I see it, this code does the right thing, but couldn't we formulate it as a one liner in Python?